### PR TITLE
Support unsigned long in sqrt and log10 for ESQL 

### PR DESCRIPTION
**Pull request by kkrik-es**
Add support for unsigned long fields in these functions. The underlying Java functions consume doubles as input, so the unsigned long values are converted to double.

Related to #98544

### CODE REVIEW

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Log10.java [hunk 1]
```diff
@@ -60,6 +64,12 @@ static double process(long val) {
         return Math.log10(val);
     }
 
+    @Evaluator(extraName = "UnsignedLong")
+    static double processUnsignedLong(long val) {
+        Number ul = NumericUtils.unsignedLongAsNumber(val);
+        return Math.log10(ul.doubleValue());
```
Comment (bpintea) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Log10.java:70]:
```suggestion
        return Math.log10(unsignedLongToDouble(val));
```

**Review by bpintea — COMMENTED** [review]
We'll want to avoid objects creation and work with primitives only, so we can use a different utility that converts ULs "directly" to double. It otherwise looks good.

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Log10.java (hunk 1 above)
Comment (kkrik-es) [reply x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Log10.java]: Done.

**Review by bpintea — APPROVED** [review]
LGTM

**Pull request merged**